redislock.go

```diff
@@ -160,13 +182,16 @@ func (l *Lock) Metadata() string {
 
 // TTL returns the remaining time-to-live. Returns 0 if the lock has expired.
 func (l *Lock) TTL(ctx context.Context) (time.Duration, error) {
-	res, err := luaPTTL.Run(ctx, l.client, []string{l.key}, l.value).Result()
-	if err == redis.Nil {
-		return 0, nil
-	} else if err != nil {
+	if l == nil {
+		return 0, ErrLockNotHeld
+	}
+	res, err := luaPTTL.Run(ctx, l.client, l.keys, l.value).Result()
+	if err != nil {
+		if err.Error() == ErrLockNotHeld.Error() {
+			return 0, ErrLockNotHeld
```

Comment:
This handling error is more explicit that `redis.Nil` but also not really necessary. Please let me know what you think about it.